modules/os2dagsorden/os2dagsorden_access_helper/inc/os2dagsorden_access_helper.permissions.inc

```diff
@@ -482,3 +481,114 @@ function os2dagsorden_access_helper_check_user_can_see_speaker_paper($sp_id, $in
 
   return FALSE;
 }
+
+/**
+ * Checks is given meeting i external committee can bee seen by the user.
+ *
+ * The function checks if user is participants of given meeting then he can see meeting content.
+ *
+ * @param object $meeting
+ *   The meeting object.
+ *
+ * @return bool
+ *   True of false.
+ */
+function os2dagsorden_access_helper_check_user_can_see_meeting_in_external_committee($meeting) {
+  $user = os2dagsorden_access_helper_get_user();
+  $access_granted = FALSE;
+  $internal_participants = field_get_items('node', $meeting, 'field_os2web_meetings_partic_int');
+  $external_participants = field_get_items('node', $meeting, 'field_os2web_meetings_partic_ext');
+  $days_before_meeting_date = (strtotime($meeting->field_os2web_meetings_date['und'][0]['value']) - time()) / 60 / 60 / 24;
+  if ($internal_participants || $external_participants) {
+    if ($internal_participants) {
+      foreach ($internal_participants as $participant) {
+        if ((int) $participant['uid'] === (int) $user->uid) {
+          if ($days_before_meeting_date < variable_get('os2dagsorden_timespan_open_meeting_content', 14)) {
+            return TRUE;
+          }
+        }
+      }
+    }
+    if ($external_participants) {
+      foreach ($external_participants as $participant) {
+        if ((int) $participant['uid'] === (int) $user->uid) {
+          if ($days_before_meeting_date < variable_get('os2dagsorden_timespan_open_meeting_content', 14)) {
+            return TRUE;
+          }
+        }
+      }
+    }
+  }
```

Comment:
instead of having two loops, you can combine internal and external participants into single array, and loop through it once.